stdlib/bytes.ml

```diff
@@ -368,3 +368,86 @@ let of_seq i =
        incr n)
     i;
   sub !buf 0 !n
+
+(** {6 Binary encoding/decoding of integers} *)
+
+external get_uint16_ne : bytes -> int -> int = "%caml_bytes_get16"
+external get_int32_ne : bytes -> int -> int32 = "%caml_bytes_get32"
+external get_int64_ne : bytes -> int -> int64 = "%caml_bytes_get64"
+external set_int16_ne : bytes -> int -> int -> unit = "%caml_bytes_set16"
+external set_int32_ne : bytes -> int -> int32 -> unit = "%caml_bytes_set32"
+external set_int64_ne : bytes -> int -> int64 -> unit = "%caml_bytes_set64"
+external swap16 : int -> int = "%bswap16"
+external swap32 : int32 -> int32 = "%bswap_int32"
+external swap64 : int64 -> int64 = "%bswap_int64"
+
+let get_uint8 b i =
+  Char.code (get b i)
+
+let get_int8 b i =
+  ((get_uint8 b i) lsl (Sys.int_size - 8)) asr (Sys.int_size - 8)
+
+let get_uint16_le b i =
+  if Sys.big_endian then swap16 (get_uint16_ne b i)
+  else get_uint16_ne b i
+
+let get_uint16_be b i =
+  if not Sys.big_endian then swap16 (get_uint16_ne b i)
+  else get_uint16_ne b i
+
+let get_int16_ne b i =
+  ((get_uint16_ne b i) lsl (Sys.int_size - 16)) asr (Sys.int_size - 16)
+
+let get_int16_le b i =
+  ((get_uint16_le b i) lsl (Sys.int_size - 16)) asr (Sys.int_size - 16)
+
+let get_int16_be b i =
+  ((get_uint16_be b i) lsl (Sys.int_size - 16)) asr (Sys.int_size - 16)
+
+let get_int32_le b i =
+  if Sys.big_endian then swap32 (get_int32_ne b i)
+  else get_int32_ne b i
+
+let get_int32_be b i =
+  if not Sys.big_endian then swap32 (get_int32_ne b i)
+  else get_int32_ne b i
+
+let get_int64_le b i =
+  if Sys.big_endian then swap64 (get_int64_ne b i)
+  else get_int64_ne b i
+
+let get_int64_be b i =
+  if not Sys.big_endian then swap64 (get_int64_ne b i)
+  else get_int64_ne b i
+
+let set_int8 b i x =
+  set b i (Char.unsafe_chr x)
```

Comment:
Do you mean `Char.unsafe_chr (x land 0xff)`?   I think the current code is ok, since unsafe_chr is `%identity` and the primitive truncates to the least significant byte.  The extra `land 0xFF` is probably not very costly; do you think I should add it?   Alternatively, I could expose directly
````
external set_int8 : bytes -> int -> int -> unit = "%bytes_safe_set"
````
(i.e. the same external as for Bytes.set, but with an `int` argument instead of `char`) to be coherent with other primitives.  Let me know what you prefer.

Reply:
> Do you mean `Char.unsafe_chr (x land 0xff)`? I think the current code is ok, since unsafe_chr is `%identity` and the primitive truncates to the least significant byte.

Yes that's what I meant. I guess you are right but OTOH you are clearly violating the expectations of `Char.unsafe_chr` and injecting values in the type that should not exist there.

Maybe I'm a bit too pedantic since this injection is only transient but simply using `%bytes_safe_set` makes things cleaner without being abusive to `Char.t`.

Reply:
Ok, see latest commit.  I'm now using directly %-primitives for int8 versions, which is more regular.